packages/react-ui/src/widgets/legend/LegendProportion.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { Box, Grid, makeStyles, Typography } from '@material-ui/core';
+
+const useStyles = makeStyles((theme) => ({
+  circles: {
+    position: 'relative',
+    marginBottom: 4
+  },
+  avg: {
+    width: 12,
+    height: 2,
+    position: 'absolute',
+    // TODO change color
+    border: `1px solid ${theme.palette.grey[900]}`,
+    borderRadius: 4,
+    backgroundColor: theme.palette.grey[900],
+    right: 0,
+    boxSizing: 'border-box'
+  }
+}));
+
+export default function LegendProportion({ legend }) {
+  const classes = useStyles();
+
+  const { labels, stats } = legend;
+
+  let max, min;
+  if (stats) {
+    min = stats.min;
+    max = stats.max;
+  } else {
+    min = labels[0];
+    max = labels[labels.length - 1];
+  }
+
+  const gap = (max + min) / 4;
+  const step1 = min + gap;
+  const step2 = max - gap;
```

Comment:
Maybe we could introduce two small functions here (inline or outside, as they are pure), like:

```js
const calculateRange = ({ labels, stats }) => {
  let max, min;
  if (stats) {
    min = stats.min;
    max = stats.max;
  } else {
    min = labels[0];
    max = labels[labels.length - 1];
  }
  return [min, max]
}

const range = calculateRange(legend);

const calculateSteps = ( [min, max] ) => {
  const gap = (max + min) / 4;
  const step1 = min + gap;
  const step2 = max - gap;

  return [step1, step2]:
}

const [step1, step2] = calculateSteps(range);

const [min, max] = range;
```